### PR TITLE
Adds Fibbonacci and Tests

### LINKED ISSUE #340: Lab1- Fibbonacci

### PR DESCRIPTION
**Pull request by mohitnag**
Fixes #340 

Review of colleague's PR #

#### Changes proposed in this PR:
- Adds the Fibonacci Function and its tests

### CODE REVIEW

File: 01_fib/mohitnag/main_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func TestMain(t *testing.T) {
+	assert := assert.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := "1,1,2,3,5,8,13"
+	actual := buf.String()
+	actual = strings.TrimRight(strings.ReplaceAll(actual, "\n", ","), ",")
```
Comment (golangcibot) [inline 01_fib/mohitnag/main_test.go]: ReplaceAll not declared by package strings (from `typecheck`)

File: 01_fib/mohitnag/main_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+func TestMain(t *testing.T) {
+	assert := assert.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := "1,1,2,3,5,8,13"
+	actual := buf.String()
+	actual = strings.ReplaceAll(actual, "\n", ",")
```
Comment (golangcibot) [inline 01_fib/mohitnag/main_test.go]: ReplaceAll not declared by package strings (from `typecheck`)

**Comment by davidbroughsmyth** [comment]
Overall looks good with a range of tests. Using slices is a different way but the outcome is the same. 
My only observation is to check for boundary conditions like when n is > 92, int will overflow. I'm not asking to make a change unless you wish perfection.

File: 01_fib/mohitnag/main.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+func fibonacci(n int) []int {
+	slice := []int{}
+	if n < 1 {
+		return []int{}
+	}
+	if n == 1 {
+		return []int{1}
+	}
+	slice = append(slice, 1, 1)
```
Comment (juliaogris) [inline 01_fib/mohitnag/main.go]: you already know how big slice has to be, so it's nicer to pre-allocate the space.
You could use something like:
```go
	if n < 1 {
		return nil
	}
	s := make([]int, n+1)
	s[0], s[1] = 0, 1
	for i := 2; i <= n; i++ {
		s[i] = s[i-1] + s[i-2]
	}
	return s[1:]
```

File: 01_fib/mohitnag/main.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	for i := 2; i < n; i++ {
+		slice = append(slice, slice[i-1]+slice[i-2])
+	}
+	return slice
+}
+
+func fib(n int) {
+	slice := fibonacci(n)
+	for i := range slice {
+		fmt.Fprintln(out, slice[i])
```
Comment (juliaogris) [inline 01_fib/mohitnag/main.go:30]: 👍 really like how you separate printing from calculating.

File: 01_fib/mohitnag/main_test.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFibonacci(t *testing.T) {
+	assert := assert.New(t)
+	testCases := []struct {
+		number   int
+		expected []int
+	}{
+		{7, []int{1, 1, 2, 3, 5, 8, 13}},
```
Comment (juliaogris) [inline 01_fib/mohitnag/main_test.go:17]: great work with the test cases.
Optional: if you want to be super perfectionist you could use a map and the key as a "sub test" name.
Here is a really good post explaining the details of that method:
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

File: 01_fib/mohitnag/main_test.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+func TestMain(t *testing.T) {
+	assert := assert.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := "1,1,2,3,5,8,13"
+	actual := buf.String()
+	actual = strings.Replace(actual, "\n", ",", -1)
```
Comment (juliaogris) [inline 01_fib/mohitnag/main_test.go:35]: could use `strings.ReplaceAll` and just "1,1,2,3,5,8,13," (extra comma at end) and skip the trim.

**Review by juliaogris — CHANGES_REQUESTED** [review]
Please [fix commit messages](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages) with `git rebase -i HASH`.

Come and see me or Cam near 1C.19 if you need any help with it.

**Comment by juliaogris** [comment]
Also, please add a review of a colleagues PR to your own PR description.

File: 01_fib/mohitnag/main.go (hunk 3 above)
Comment (mohitnag) [reply 01_fib/mohitnag/main.go]: Done

File: 01_fib/mohitnag/main_test.go (hunk 6 above)
Comment (mohitnag) [reply 01_fib/mohitnag/main_test.go:35]: Done

**Review by juliaogris — APPROVED** [review]
lgtm
